compiler/src/dotty/tools/repl/Rendering.scala

```diff
@@ -131,7 +137,13 @@ private[repl] class Rendering(parentClassLoader: Option[ClassLoader] = None):
    */
   private def rewrapValueClass(sym: Symbol, value: Object)(using Context): Option[Object] =
     if ValueClasses.isDerivedValueClass(sym) then
-      val valueClassName = sym.flatName.encode.toString
+      val pkg = sym.enclosingPackageClass
+      val pkgName = if pkg.isEmptyPackage then "" else s"${pkg.fullName.mangledString}."
+      val clsFlatName = if sym.isOneOf(JavaDefined | ConstructorProxy) then
+        // See ExtractDependencies.recordDependency
```

Comment:
I would prefer if we could extract the logic out of recordDependency in a shared method, it looks like we have a potential third user for this method too: https://github.com/lampepfl/dotty/pull/16392#discussion_r1036892345